types/jsonwebtoken/jsonwebtoken-tests.ts

```diff
@@ -142,6 +142,10 @@ jwt.verify(token, cert, { ignoreExpiration: true }, (err, decoded) => {
     // if ignoreExpration == false and token is expired, err == expired token
 });
 
+// verify a non-signed token
+const unsigned = jwt.sign({ aud: 'foo', sub: 'bar' }, null, { algorithm: 'none' });
+jwt.verify(unsigned, null, { algorithms: ['none' ]});
```

Comment:
I've also tried the exact test from the commit:
https://github.com/auth0/node-jsonwebtoken/blob/master/test/verify.tests.js#L44
```ts
jwt.verify(signed, null, {typ: 'JWT'}, function(err, p) {
..
```
this use-case does not work with the TS usage. I don't know internals much of this package,
to have a clear stance on what to do, thoughts?

Reply:
@peterblazejewicz that's a good point ... that example has multiple problems because `typ` is not even a property of `VerifyOptions` today.

Suggestions?

Reply:
I'd align with code we know (from original source). In the meantime, maybe we will got more input:
@SomaticIT @danielheim @brikou @vpk @kettil @RunAge @nflaig @ivansieder